scripts/gluster-block/setup-gluster-block-glusto.yml

```diff
@@ -0,0 +1,392 @@
+      #baseurl: "{{ 'http://' + hostvars[groups['gluster_nodes'][0]]['ansible_eth0']['ipv4']['address'] + '/RPMS/' }}"
+      baseurl: http://artifacts.ci.centos.org/gluster/nightly/master/$releasever/$basearch
+      gpgcheck: no
+      repo_gpgcheck: no
+
+  - name: Install gluster
+    yum: name={{ item }} state=installed
+    with_items:
+    - glusterfs-server
+    - glusterfs-cli
+    - glusterfs-debuginfo
+
+  - name: Install gluster GNFS package
+    yum: name=glusterfs-gnfs state=installed
+    ignore_errors: True
+
+  - name: Allocate space to a file
+    command: "fallocate -l 50G /var/{{item}}"
+    args:
+      creates: "/var/{{item}}"
+    with_items:
+    - data0
+    - data1
+    - data2
+    - data3
+
+  - name: Associate loopback device to file
+    command: "losetup -f /var/{{item.1}}"
+    args:
+      creates: "/dev/{{item.0}}"
+    with_together:
+    - ['loop0', 'loop1', 'loop2', 'loop3']
+    - ['data0', 'data1', 'data2', 'data3']
+    loop_control:
+      pause: 2
+    ignore_errors: True
+
+  - name: Create physical volume
+    command: "pvcreate /dev/{{item}}"
+    with_items:
+    - loop0
+    - loop1
+    - loop2
+    - loop3
+    loop_control:
+      pause: 2
+    ignore_errors: True
+
+  - name: Create volume groups
+    lvg:
+      vg: "{{item.0}}"
+      pvs: "/dev/{{item.1}}"
+    ignore_errors: yes
+    with_together:
+    - ['group0', 'group1', 'group2', 'group3']
+    - ['loop0', 'loop1', 'loop2', 'loop3']
+
+  - name: Create thin pool
+    command: "lvcreate -L 45G -T {{item}}/pool"
+    args:
+      creates: "/dev/{{item}}"
+    ignore_errors: yes
+    with_items:
+    - group0
+    - group1
+    - group2
+    - group3
+
+  - name: Create thin volume
+    command: "lvcreate -V 45G -T {{item.0}}/pool -n {{item.1}}"
+    args:
+      creates: "/dev/{{item.0}}/{{item.1}}"
+    ignore_errors: yes
+    with_together:
+    - ['group0', 'group1', 'group2', 'group3']
+    - ['vol0', 'vol1', 'vol2', 'vol3']
+
+  - name: Format the volumes
+    command: "mkfs.xfs /dev/{{item.0}}/{{item.1}}"
+    ignore_errors: yes
+    with_together:
+    - ['group0', 'group1', 'group2', 'group3']
+    - ['vol0', 'vol1', 'vol2', 'vol3']
+
+  - name: Mount the bricks
+    mount: name="/mnt/{{item.0}}" src="/dev/{{item.1}}" state=mounted fstype=xfs
+    with_together:
+    - ['vol0', 'vol1', 'vol2', 'vol3']
+    - ['group0/vol0', 'group1/vol1', 'group2/vol2', 'group3/vol3']
+
+  - name: Start Gluster Daemon
+    service: name=glusterd state=started
+
+  - name: Status rpcbind
+    service: name=rpcbind state=started
+
+
+#Set up gluster-block and related packages on servers
+- hosts: gluster_nodes[1:4]
+  tasks:
+  - name: Install dependency packages
+    yum: name={{ item }} state=installed
+    with_items:
+    - git
+    - libtool
+    - libuuid-devel
+    - python-gobject
+    - glusterfs-api-devel
+    - automake
+    - autoconf
+    - gcc
+    - glib2-devel
+    - "*kmod*"
+    - libnl3*
```

Comment:
Same question

Reply:
Because there are multiple packages ending with kmod and starting with libnl3 and we need all of them for block installation.